File: Java/src/main/java/Solution1001_1100/Solution1021.java
package Solution1001_1100;

import java.util.Stack;

/**
 * @author windows
 */
public class Solution1021 {
    public String removeOuterParentheses(String s) {
        int left = 0;
        Stack stack = new Stack();
        String res = "";
        for (int i = 0; i < s.length(); i++) {
            if (s.charAt(i) == '('){
                stack.push('(');
            }else{
                stack.pop();
            }
            if (stack.isEmpty()) {
                res += s.substring(left + 1, i);
                left = i + 1;
            }
        }
        return res;
    }


    
    public static void main(String[] args) {
        Solution1021 solution1021 = new Solution1021();
    }
}

File: C++/src/Solution1001_1100/Solution1021.cpp
//
// Created by windows on 2023/3/2.
//

#include "Solution1021.h"
#include <iostream>
#include <vector>
#include <unordered_set>
#include <unordered_map>
#include <string>
using namespace std;

string removeOuterParentheses(string s) {
    string res;
    int left = 0;
    int right = 0;
    int begin = 0;
    for (int i = 0; i < s.size(); i++) {
      if (s[i] == '(') {
        left++;
      }else{
        right++;
      }
      if (left == right){
        res.append(s.substr(begin + 1, i - begin - 1));
        begin = i + 1;
      }
    }
    return res;
}